src/api/Controllers/PaginatedResponse.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace BDMS.Controllers;
+
+/// <summary>
+/// Represents a paginated response containing a subset of items along with pagination info.
+/// </summary>
+/// <typeparam name="T">The type of the items contained in the paginated response.</typeparam>
+public class PaginatedResponse<T>
```

Comment:
Q/REQ: Muss dieser Typ generisch sein? Aktuell wird es nur für Boreholes verwendet und in naher Zukunft vermutlich für nichts anderes. Falls dem so ist, würde ich den Typen zu z.B. `PaginatedBoreholesResponse` oder `GetAllBoreholesResponse` umbenennen und den generischen Parameter weglassen.  
Zusätzlich zu prüfen: `record`-Type anstatt `class` verwenden. Mit `record` spart man sich den Konstruktor unten.